internal/provider/provider_framework.go

```diff
@@ -0,0 +1,194 @@
+		// the provider handled used the single organization's single project by default, instead of requiring the
+		// user to set it. Once multiple projects are available, this helper issues a warning: when multiple projects exist within the org,
+		// a project ID should be set on the provider or on each resource. Otherwise, the oldest project will be used by default.
+		// This helper will eventually be deprecated after a migration period.
+		project, projDiags := getProjectFromCredentialsFramework(ctx, client)
+		if projDiags != nil {
+			if !projDiags.HasError() {
+				resp.Diagnostics.Append(projDiags...)
+			} else {
+				resp.Diagnostics.AddError("unable to get project from credentials", "")
+				return
+			}
+		}
+
+		client.Config.OrganizationID = project.Parent.ID
+		client.Config.ProjectID = project.ID
+	}
+
+	var config ProviderFrameworkConfiguration
+	config.Client = client
+	resp.DataSourceData = client
+	resp.ResourceData = client
+}
+
+// getProjectFromCredentials uses the configured client credentials to
+// fetch the associated organization and returns that organization's
+// single project.
+func getProjectFromCredentialsFramework(ctx context.Context, client *clients.Client) (project *models.HashicorpCloudResourcemanagerProject, diags diagnostic.Diagnostics) {
+	// Get the organization ID.
+	listOrgParams := organization_service.NewOrganizationServiceListParams()
+	listOrgResp, err := clients.RetryOrganizationServiceList(client, listOrgParams)
+	if err != nil {
+		diags.AddError(fmt.Sprintf("unable to fetch organization list: %v", err), "")
+
+		return nil, diags
+	}
+	orgLen := len(listOrgResp.Payload.Organizations)
+	if orgLen != 1 {
+		diags.AddError(fmt.Sprintf("unexpected number of organizations: expected 1, actual: %v", orgLen), "")
+		return nil, diags
+	}
+	orgID := listOrgResp.Payload.Organizations[0].ID
+
+	// Get the project using the organization ID.
+	listProjParams := project_service.NewProjectServiceListParams()
+	listProjParams.ScopeID = &orgID
+	scopeType := string(models.HashicorpCloudResourcemanagerResourceIDResourceTypeORGANIZATION)
+	listProjParams.ScopeType = &scopeType
+	listProjResp, err := clients.RetryProjectServiceList(client, listProjParams)
+	if err != nil {
+		diags.AddError(fmt.Sprintf("unable to fetch project id: %v", err), "")
+		return nil, diags
+	}
+	if len(listProjResp.Payload.Projects) > 1 {
+		diags.AddWarning("There is more than one project associated with the organization of the configured credentials.", `The oldest project has been selected as the default. To configure which project is used as default, set a project in the HCP provider config block. Resources may also be configured with different projects.`)
+
+		return getOldestProject(listProjResp.Payload.Projects), diags
+	}
+	project = listProjResp.Payload.Projects[0]
+	return project, diags
+}
+
+func (p *ProviderFramework) Resources(ctx context.Context) []func() resource.Resource {
```

Comment:
Code organization-wise, I would move this up, after the Schema definition, so it's easy to see the list of supported resources.

Reply:
Totally fair!